lib/puppet/property.rb

```diff
@@ -55,6 +55,10 @@ class Puppet::Property < Puppet::Parameter
   #
   attr_writer :noop
 
+  # @!attribute [rw] sensitive
+  #   @return [true, false] If this property has been tagged as sensitive.
+  attr_accessor :sensitive
```

Comment:
Branan's comments are why I didn't push this to the `Puppet::Parameter` class; eventually when we pass instances of the `Sensitive` type to the RAL it may be useful - but at that point we won't need an attribute on the Parameter class and will handle the value on a per-parameter value basis.